### ISSUE TITLE
review 

### ISSUE THREAD

**Issue by arademaker** [LR-POR/ud-matrix#3]
as analises UD, principalmente do arquivo examples.conllu que estão vindo de https://github.com/LR-POR/PorGram/issues/19#issuecomment-902317884 precisam de revisão. @leoalenc se puder ajudar...

**Comment by leoalenc** [comment]
> as analises UD, principalmente do arquivo examples.conllu que estão vindo de [LR-POR/PorGram#19 (comment)](https://github.com/LR-POR/PorGram/issues/19#issuecomment-902317884) precisam de revisão. @leoalenc se puder ajudar...

@arademaker, sim, gostaria de ajudar. Onde estão mesmo os exemplos a ser verificados?

**Comment by arademaker** [comment]
https://github.com/LR-POR/ud-matrix/blob/master/examples.conllu

a idéia é comecar a usar este repositório (talvez mudar o nome) como o local para os testsets e profiles da PorGram. E podemos tentar sempre fazer com que todos os exemplos tenham analise UD também.

**Comment by leoalenc** [comment]
> https://github.com/LR-POR/ud-matrix/blob/master/examples.conllu
> 
> a idéia é comecar a usar este repositório (talvez mudar o nome) como o local para os testsets e profiles da PorGram. E podemos tentar sempre fazer com que todos os exemplos tenham analise UD também.

@arademaker, vou verificar os exemplos.

**Comment by arademaker** [comment]
OK, fiz uma revisão, alguns casos para confirmar:

1. obj vs iobj como `queixou-se do cachorro` seria iobj ou obj? 
2. xcomp vs ccomp  `queria dormir` xcomp? mas 'o gato queria o que? dormir` certo? me parece ccomp pensando assim. As guidelines não são claras neste caso.

**Comment by leoalenc** [comment]
> OK, fiz uma revisão, alguns casos para confirmar:
> 
> 1. obj vs iobj como `queixou-se do cachorro` seria iobj ou obj?

@arademaker, *obj*. 

> 2. xcomp vs ccomp  `queria dormir` xcomp? mas 'o gato queria o que? dormir` certo? me parece ccomp pensando assim. As guidelines não são claras neste caso.

@arademaker, *xcomp*, isso para mim está bastante claro. É exemplo clássico de *xcomp*. 
 
@arademaker , nesses poucos exemplos, deparamo-nos com questões que impactam toda a revisão do Bosque. Isso mostra, ao meu ver, um lado interessante do exercício. Temos uma proposta de metodologia de revisão de *treebank* que usa as definições de uma gramática, no caso referentes à valência, para dar consistência nas anotações. O que torna o exercício desafiador é que as teorias subjacentes são diferentes. Na HPSG, diferentemente da LFG e UD, as funções gramaticais parecem não ter muita importância. 
Minha atitude agora é a seguinte, baseada no que li da UD (que muito bebe da teoria da LFG):
1. *obj* seria o objeto direto, sem preposição, portanto. Desse modo, em 81, o nó 6 seria *obl*. Em 91, *cachorro* também seria *obl*. Mas vejo que UD trata esses casos como *[obj](https://universaldependencies.org/u/dep/obj.html)*.
2. *[iobj](https://universaldependencies.org/u/dep/iobj.html)* é o objeto indireto no sentido de *objeto secundário* da LFG, ou seja, só teríamos *iobj* se o verbo reger também um *obj*.

> In general, if there is just one object, it should be labeled obj, regardless of the morphological case or semantic role that it bears. If there are two or more objects, one of them should be obj and the others should be iobj. 
https://universaldependencies.org/u/dep/obj.html

3. No caso de *verba dicendi*, como *contar*, *dizer* etc., quando o argumento correspondente à mensagem se realiza como *ccomp* ou *xcomp*, eu também acho que faz sentido dizer que o recipiente se realiza como *iobj*. Existe uma analogia forte entre transmissão de mensagem e transferência de posse, caso prototípico de verbo ditransitivo. Além disso, a mensagem ou proposição é realizável como *obj*: *eu o contei a ela* ou *eu contei-lho* no português europeu. Assim, sugiro que *estudante* em 111 seja *iobj*.

Seria importante marcar o reflexivo como tal, até mesmo marcando como [expletivo](https://universaldependencies.org/u/dep/expl.html), mais especificamente *expl:pv* no caso de 124.

**Comment by leoalenc** [comment]
@arademaker, veja correção:

```
sent_id = 124
porgram = X-compl-subj-raise-aux-with-pred
text = O cachorro tem que dormir.
1	O	o	DET	_	Definite=Def|Gender=Masc|Number=Sing|PronType=Art	2	det	_	_
2	cachorro	cachorro	NOUN	_	Gender=Masc|Number=Sing	5	nsubj	_	_
3	tem	ter	AUX	_	Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin	5	aux	_	_
4	que	que	SCONJ	_	_	5	mark	_	_
5	dormir	dormir	VERB	_	VerbForm=Inf	0	root	_	SpaceAfter=No
6	.	.	PUNCT	_	_	5	punct	_	SpaceAfter=No
```

3 não é AUX, mas VERB, sendo 5 o seu *xcomp*. Compare:

>O cachorro precisa dormir.
>O cachorro tem de dormir.

**Comment by leoalenc** [comment]
@arademaker, https://github.com/LR-POR/ud-matrix/pull/5/commits/2d47b0613d93b2cf6d715454eb4193f21fdacc1c deve fechar _issue_ se você concordar com as alterações.

**Comment by arademaker** [comment]
só não entendi o que vc quis destacar quando pede para comparar o exemplos:

1. O cachorro precisa dormir.
2. O cachorro tem de dormir.

**Comment by arademaker** [comment]
> 1. Assim, sugiro que _estudante_ em 111 seja _iobj_.

Vc mudou de opinião ou esqueceu este ajuste na sua revisão no PR #5 ? Por que estudante permanece como `obj`.

**Comment by arademaker** [comment]
@leoalenc estes dois últimos comentários são para você... por favor, para fecharmos este issue.

**Comment by leoalenc** [comment]
> só não entendi o que vc quis destacar quando pede para comparar o exemplos:
> 
> 1. O cachorro precisa dormir.
> 2. O cachorro tem de dormir.

@arademaker , compare sent_id = 124:

>O cachorro tem que dormir. 

com 

> 1. O cachorro precisa dormir.
> 2. O cachorro tem de dormir.

e 

> O menino precisa concluir a tarefa.
> O menino tem de concluir a tarefa.
> O menino tem que concluir a tarefa.
> A tarefa precisa ser concluída pelo menino.
> A tarefa tem que ser concluída pelo menino.
> A tarefa tem de ser concluída pelo menino.


Isso mostra que as estruturas desses três verbos precisam ser paralelas, com a única diferença de que _precisar_ não exige complementador (e as duas variantes de _ter_ exigem complementadores diferentes). Trata-se de verbos de alçamento do sujeito, conforme Sag, Wasow e Bender. Ver:

https://github.com/LR-POR/PorGram/issues/53

https://github.com/LR-POR/PorGram/issues/36

**Comment by leoalenc** [comment]
> > 1. Assim, sugiro que _estudante_ em 111 seja _iobj_.
> 
> Vc mudou de opinião ou esqueceu este ajuste na sua revisão no PR #5 ? Por que estudante permanece como `obj`.

@arademaker, isso depende da discussão desta _issue_:

https://github.com/UniversalDependencies/UD_Portuguese-Bosque/issues/360

Se o objetivo é extrair as molduras de valência de verbos ditransitivos que regem um complemento oracional (`ccomp` ou `xcomp`), então deveremos procurar tanto por `<nsubj obj ccomp>` quanto `<nsubj iobj ccomp>`, por exemplo. 
Em suma, só podemos fechar a presente questão quando resolvermos a questão acima do Bosque. Eu sugeri `iobj` com base nas minha convicções sintáticas, mas vi inicialmente que o corpus anota esse tipo de constituinte como `obj`, portanto, não corrigi o _output_ do _udpipe_. Agora sabemos que o corpus tem também `iobj` na mesma configuração, conforme destaquei em https://github.com/UniversalDependencies/UD_Portuguese-Bosque/issues/360.
